src/cascadia/TerminalApp/AppKeyBindings.cpp

```diff
@@ -1,117 +1,294 @@
-    DEFINE_EVENT(AppKeyBindings, CloseWindow,       _CloseWindowHandlers,       TerminalApp::CloseWindowEventArgs);
-    DEFINE_EVENT(AppKeyBindings, CloseTab,          _CloseTabHandlers,          TerminalApp::CloseTabEventArgs);
-    DEFINE_EVENT(AppKeyBindings, SwitchToTab,       _SwitchToTabHandlers,       TerminalApp::SwitchToTabEventArgs);
-    DEFINE_EVENT(AppKeyBindings, NextTab,           _NextTabHandlers,           TerminalApp::NextTabEventArgs);
-    DEFINE_EVENT(AppKeyBindings, PrevTab,           _PrevTabHandlers,           TerminalApp::PrevTabEventArgs);
-    DEFINE_EVENT(AppKeyBindings, IncreaseFontSize,  _IncreaseFontSizeHandlers,  TerminalApp::IncreaseFontSizeEventArgs);
-    DEFINE_EVENT(AppKeyBindings, DecreaseFontSize,  _DecreaseFontSizeHandlers,  TerminalApp::DecreaseFontSizeEventArgs);
-    DEFINE_EVENT(AppKeyBindings, ScrollUp,          _ScrollUpHandlers,          TerminalApp::ScrollUpEventArgs);
-    DEFINE_EVENT(AppKeyBindings, ScrollDown,        _ScrollDownHandlers,        TerminalApp::ScrollDownEventArgs);
-
-
-}
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+#include "pch.h"
+#include "AppKeyBindings.h"
+
+using namespace winrt::Microsoft::Terminal;
+using namespace winrt::TerminalApp;
+using namespace winrt::Windows::Data::Json;
+
+static const std::wstring KEYS_KEY{ L"keys" };
+static const std::wstring COMMAND_KEY{ L"command" };
+
+static const std::wstring COPYTEXT_KEY{ L"copy" };
+static const std::wstring PASTETEXT_KEY{ L"paste" };
+static const std::wstring NEWTAB_KEY{ L"newTab" };
+static const std::wstring NEWTABWITHPROFILE0_KEY{ L"newTabProfile0" };
+static const std::wstring NEWTABWITHPROFILE1_KEY{ L"newTabProfile1" };
+static const std::wstring NEWTABWITHPROFILE2_KEY{ L"newTabProfile2" };
+static const std::wstring NEWTABWITHPROFILE3_KEY{ L"newTabProfile3" };
+static const std::wstring NEWTABWITHPROFILE4_KEY{ L"newTabProfile4" };
+static const std::wstring NEWTABWITHPROFILE5_KEY{ L"newTabProfile5" };
+static const std::wstring NEWTABWITHPROFILE6_KEY{ L"newTabProfile6" };
+static const std::wstring NEWTABWITHPROFILE7_KEY{ L"newTabProfile7" };
+static const std::wstring NEWTABWITHPROFILE8_KEY{ L"newTabProfile8" };
+static const std::wstring NEWTABWITHPROFILE9_KEY{ L"newTabProfile9" };
+static const std::wstring NEWWINDOW_KEY{ L"newWindow" };
+static const std::wstring CLOSEWINDOW_KEY{ L"closeWindow" };
+static const std::wstring CLOSETAB_KEY{ L"closeTab" };
+static const std::wstring SWITCHTOTAB_KEY{ L"switchToTab" };
+static const std::wstring NEXTTAB_KEY{ L"nextTab" };
+static const std::wstring PREVTAB_KEY{ L"prevTab" };
+static const std::wstring INCREASEFONTSIZE_KEY{ L"increaseFontSize" };
+static const std::wstring DECREASEFONTSIZE_KEY{ L"decreaseFontSize" };
+static const std::wstring SCROLLUP_KEY{ L"scrollUp" };
+static const std::wstring SCROLLDOWN_KEY{ L"scrollDown" };
+
+static const std::vector<std::pair<ShortcutAction, std::wstring>> commandNames {

```

Comment:
a vector of pairs seems like a map to me

Reply:
I was trying to keep the ordering as important here, since later on we want to make sure to always serialize them in this order. 

It looks like I can keep a `map` ordered though, so I can update that.